drivers/hmis/app/models/hmis/hud/processors/client_processor.rb

```diff
@@ -17,8 +17,18 @@ def process(field, value)
         gender_attributes(Array.wrap(attribute_value))
       when 'pronouns'
         { attribute_name => Array.wrap(attribute_value).any? ? Array.wrap(attribute_value).join('|') : nil }
-      when 'SSN'
-        { attribute_name => attribute_value.present? ? attribute_value.gsub(/[^\dXx]/, '') : nil }
+      when 'ssn'
+        if value == Base::HIDDEN_FIELD_VALUE
+          { attribute_name => @processor.send(factory_name).ssn }
+        else
+          { attribute_name => attribute_value.present? ? attribute_value.gsub(/[^\dXx]/, '') : nil }
+        end
+      when 'dob'
+        { attribute_name => value == Base::HIDDEN_FIELD_VALUE ? @processor.send(factory_name).dob : attribute_value }
```

Comment:
For SSN and DOB we shouldn't need to explicitly re-set them to the old value. We can just skip them.

For the data quality fields, it looks like you ran into an issue with skipping them because they are required fields, so must be set to 99 for new clients. Kinda clunky, but it's ok.

Here's a suggestion that doesn't do the unnecessary SSN/DOB-setting, plus some comments:

```
    def process(field, value)
      attribute_name = hud_name(field)
      attribute_value = attribute_value_for_enum(hud_type(field), value)

      # Skip SSN/DOB fields if hidden, because they are always hidden due to lack of permissions (see client.json form definition)
      return if value == Base::HIDDEN_FIELD_VALUE && ['ssn', 'dob'].include?(attribute_name)

      attributes = case attribute_name
      when 'race'
        race_attributes(Array.wrap(attribute_value))
      when 'gender'
        gender_attributes(Array.wrap(attribute_value))
      when 'pronouns'
        { attribute_name => Array.wrap(attribute_value).any? ? Array.wrap(attribute_value).join('|') : nil }
      when 'ssn'
        { attribute_name => attribute_value.present? ? attribute_value.gsub(/[^\dXx]/, '') : nil }
      when 'ssn_data_quality'
        # If hidden due to permissions, set to old value or 99
        attribute_value = @processor.send(factory_name).ssn_data_quality || 99 if value == Base::HIDDEN_FIELD_VALUE
        { attribute_name => attribute_value }
      when 'dob_data_quality'
        # If hidden due to permissions, set to old value or 99
        attribute_value = @processor.send(factory_name).dob_data_quality || 99 if value == Base::HIDDEN_FIELD_VALUE
        { attribute_name => attribute_value }
      else
        { attribute_name => attribute_value }
      end

      @processor.send(factory_name).assign_attributes(attributes)
    end
```